packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -895,10 +904,30 @@ export abstract class FluidDataStoreContext
 	 * other clients that are up-to-date till seq# 100 may not have them yet.
 	 */
 	protected identifyLocalChangeInSummarizer(eventName: string, type?: string) {
-		if (
-			this.clientDetails.type !== summarizerClientType ||
-			this.localChangesTelemetryCount <= 0
-		) {
+		if (this.clientDetails.type !== summarizerClientType) {
+			return;
+		}
+
+		if (this.abortSummarizerIfLocalChanges) {
+			const error = DataProcessingError.create(
+				"Summarizer with local changes - DataStoreContext",
+				"unexpectedActionReceived",
+				undefined,
+				{
+					eventName,
+					type,
```

Comment:
How about `messageType`?  Also consider putting this and `isSummaryInProgress` as JSON in a property `errorDetails`.

Reply:
I changed the messageType - not sure I want to move it to the error details due to the fact that we already have similar logging that does not.